docs/en/query_language/dicts/external_dicts_dict_hierarchical.md

```diff
@@ -0,0 +1,35 @@
+# Hierarchical Dictionaries
+
+ClickHouse supports hierarchical dictionaries with a [numeric key](external_dicts_dict_structure.md#ext_dict-numeric-key).
+
+Look at the following hierarchical structure:
+
+```text
+0 (Common ancestor)
+│
+├── 1 (Russia)
+│   │
+│   └── 2 (Moscow) 
+│       │
+│       └── 3 (Center)  
+│
+└── 4 (Great Britain)
+    │
+    └── 5 (London)
+```
+
+This hierarchy can be expressed as the following dictionary table.
+
+Key | Ancestor key | Name
+----|--------------|------
+1 | 0 | Russia
+2 | 1 | Moscow
+3 | 2 | Center
+4 | 0 | Great Britain
+5 | 4 | London
+
+This table contains an attribute that equals the key of the nearest ancestor for the element.
+
+You can configure an hierarchical dictionary by setting the [hierarchical] (external_dict_dict_structure.md#hierarchical-dict-attr) property for the attribute that must contain the ancestor key.
```

Comment:
I can't understand the logic. And it is too complicated.
configure..setting..property..attribute..must contain..ancestor.

Probably better tell it in several sentences. To enable ...  you should do following: 
- Table should have ancestor column with name `zzzz`.
- Parameter xxx is set to true.

Actually I can't tell how I should name columns in table for hierarchical dict to work. I don't see exact name here and there is no such parameter.

Reply:
I've rewritten all this part.